extensions/wikia/WallNotifications/WallNotifications.class.php

```diff
@@ -30,70 +30,75 @@ public function __construct() {
 	/*
 	 * Public Interface
 	 */
-	public function getWikiNotifications( $userId, $wikiId, $readSlice = 5, $countonly = false, $notifyeveryone = false ) {
-		/* if since == null, get all notifications */
-		/* possibly ignore $wiki at one point and fetch notifications from all wikis */
-		wfProfileIn(__METHOD__);
-
-		$memcSync = $this->getCache( $userId, $wikiId );
-
-		// try fetching the list of notifications from memcache
-		$list = $memcSync->get();
-		if ( empty( $list ) && !is_array( $list ) ) {
-			// nothing in the cache, so use the db as a fallback and store the result in cache
-			$callback = function() use( $userId, $wikiId, &$list ) {
-				$list = $this->rebuildData( $userId, $wikiId, false );
-				return $list;
-			};
 
-			// this memcache data is synchronized so we're making sure nothing else is modifying it at the same time
-			// we're using the same callback when we cannot acquire the lock, as we want to have the list of notifications
-			// even if we won't be able to store it in the cache
-			$memcSync->lockAndSetData( $callback, $callback );
-		}
+	/**
+	 * Get all notifications for a user on a wiki
+	 *
+	 * @param int $userId The user ID receiving notifications
+	 * @param int $wikiId The wikia on which the notifications originated
+	 * @param int $readSlice Max number of already read messages to show
+	 * @param bool $countOnly Set to true to only return a count.  This should be split out to another function
+	 * @param bool $notifyEveryone
+	 *
+	 * @return array|Mixed
+	 */
+	public function getWikiNotifications( $userId, $wikiId, $readSlice = 5, $countOnly = false, $notifyEveryone = false ) {
 
+		$list = $this->getWikiNotificationList( $userId, $wikiId );
 		if ( empty( $list ) ) {
```

Comment:
It seems superfluous to check empty($list) both here and in getWikiNotificationList. Why not just "return $list" in getWikiNotificationList and leave the "return []" here?